chirho/explainable/handlers/components.py

```diff
@@ -8,12 +8,47 @@
 from chirho.counterfactual.handlers.selection import get_factual_indices
 from chirho.explainable.internals import uniform_proposal
 from chirho.indexed.ops import IndexSet, gather, indices_of, scatter_n
-from chirho.observational.handlers import soft_neq
+from chirho.observational.handlers import soft_eq, soft_neq
 
 S = TypeVar("S")
 T = TypeVar("T")
 
 
+def sufficiency_intervention(
```

Comment:
I though about going this way, but then I thought that there is a division of labor there, where `sufficiency_intervention` just like `random_intervention` operates on values, whereas picking which values to operate on in terms of what the antecedents are is left to `SearchFor...` handler, as before:

```
antecedents_supports = {a: s for a, s in antecedents.items()}
        antecedents = {
            a: random_intervention(s, name=f"{antecedent_prefix}_proposal_{a}")
            for a, s in antecedents.items()
        }
  ```
  
  In general, we could move this behavior into both `random_intervention` and `sufficiency_intervention`, but wouldn't it make these less general?

Reply:
The reason I suggested this change is that there could be other index variables introduced by other unrelated handlers, for example to [vectorize sampling](https://basisresearch.github.io/chirho/robust.html#chirho.robust.internals.nmc.BatchedLatents). A list of `antecedents` names is necessary for `sufficiency_intervention` to avoid conflating the antecedent index variables of `SearchForExplanation` with those unrelated variables that may not even correspond to counterfactual `split` operation calls and should be passed through unaffected.